ui/components/MesherySettings.js

```diff
@@ -134,205 +134,147 @@ TabContainer.propTypes = { children : PropTypes.node.isRequired, };
 
 
 //TODO: Tabs are hardcoded everywhere
-class MesherySettings extends React.Component {
-  constructor(props) {
-    super(props);
-    const {
-      k8sconfig, meshAdapters, grafana, prometheus, router : { asPath }
-    } = props;
-
-    this._isMounted = false;
-
-    //TODO: Extract this logic and clean it
-    let tabVal = 0, subTabVal = 0;
-    const splittedPath = asPath.split('#');
-    if (splittedPath.length >= 2 && splittedPath[1]) {
-      const subTabPath = splittedPath[1].split('/');
-
-      switch (subTabPath[0]) {
-
-        case 'environment':
-          tabVal = 0;
-          break;
-        case 'service-mesh':
-          tabVal = 1;
-          break;
-        case 'metrics':
-          tabVal = 2;
-          break;
-        case 'system':
-          tabVal = 3;
-          break
-        case 'meshmodel-summary' :
-          tabVal = 4;
-          break
-        // case 'performance':
-        //   tabVal = 3;
-        //   break;
-      }
-      if (subTabPath.length >= 2 && subTabPath[1]) {
-        switch (subTabPath[1]) {
-          case 'inclusterconfig':
-            subTabVal = 0;
-            break;
-          case 'outclusterconfig':
-            subTabVal = 1;
-            break;
-          case 'grafana':
-            subTabVal = 0;
-            break;
-          case 'prometheus':
-            subTabVal = 1;
-            break;
-        }
-      }
-    }
-    this.state = {
-      completed : {},
-      k8sconfig,
-      meshAdapters,
-      grafana,
-      prometheus,
-      tabVal,
-      subTabVal,
-      modelsCount : 0,
-      componentsCount : 0,
-      relationshipsCount : 0,
-      isMeshConfigured : k8sconfig.clusterConfigured,
-
-      // Array of scanned prometheus urls
-      scannedPrometheus : [],
-      // Array of scanned grafan urls
-      scannedGrafana : [],
-    };
-
-    this.systemResetPromptRef = React.createRef();
-  }
+const MesherySettings = (props) => {
+  const {
+    k8sconfig,
+    meshAdapters,
+    grafana,
+    prometheus,
+    classes,
+  } = props;
+
+  const  notify  = useNotification();
+  const [modelsCount, setModelsCount] = useState(0);
+  const [componentsCount, setComponentsCount] = useState(0);
+  const [relationshipsCount, setRelationshipsCount] = useState(0);
+  const [scannedGrafana, setScannedGrafana] = useState([]);
+  const [scannedPrometheus, setScannedPrometheus] = useState([]);
+  const [tabVal, setTabVal] = useState(0);
+  const [subTabVal, setSubTabVal] = useState(0);
+  const [isMeshConfigured] = useState(k8sconfig.clusterConfigured);
```

Comment:
why are we storing this in state , it is derived from props

Reply:
@aabidsofi19 storing in state has multiple advantages like increase the Performance Optimization and code readability 
But we can use derived props too . But in functional component its recomended to use the state .